packages/core/src/lib/light_push/index.ts

```diff
@@ -51,23 +43,8 @@ class LightPush implements ILightPush {
   ): Promise<SendResult> {
     const { pubSubTopic = DefaultPubSubTopic } = this.options;
 
-    const res = await selectPeerForProtocol(
-      this.peerStore,
-      [this.multicodec],
-      opts?.peerId
-    );
-
-    if (!res) {
-      throw new Error("Failed to get a peer");
-    }
-    const { peer } = res;
-
-    const connections = this.libp2p.getConnections(peer.id);
-    const connection = selectConnection(connections);
-
-    if (!connection) throw "Failed to get a connection to the peer";
-
-    const stream = await connection.newStream(LightPushCodec);
+    const peer = await this.getPeer(opts?.peerId);
```

Comment:
I believe this can be done more optimal with `await Promise.all`

I think we have a recurring pattern here. Might make sense to create an util that will proxy `Promise.all` and improve usability. Not sure how it should look like now.

Reply:
Sorry not sure I got it. `await Promise.all` on what?

```
    const peer = await this.getPeer(opts?.peerId);
    const stream = await this.newStream(peer);
```

The actions are sequential as `getPeer` needs to resolve before `newStream` can be called. `Promise.all` usually helps with async operations that can be done in parallel.

Reply:
Yeah, seems I misread it